### PR TITLE
Feature/vdx 137/handle subject syntax types

### PR DESCRIPTION
**Pull request by sksadjad**
updates:
- removed checkedLinkedDomain property from `VerifyAuthenticationResponseOpts` and `VerifyAuthenticationRequestOpts` because we are using it from `Internal/External Verification` object
- [if we don't pass a checkLinkedDomain option to our OP/RP it will use CheckLinkedDomain.IF_PRESENT
- fixed the logic behind finding intersection of OP & RP supported `subjectSyntaxTypes`
- added another enum for `SubjectSyntaxTypesSupportedValues` that has values of `did` and `urn:ietf:params:oauth:jwk-thumbprint`
- fixed the problem with handling the `subject_syntax_types` when OP/RP are initialized using `withDidMethod`
- fixed the problem with `nonce` value in populating `AuthenticationRequestPayload`
- fixed the tests with the correct logic of finding intersection of of OP & RP supported `subjectSyntaxTypes`

### CODE REVIEW

File: src/main/AuthenticationRequest.ts [hunk 1]
```diff
@@ -276,7 +276,7 @@ function createClaimsPayload(opts: ClaimOpts): ClaimPayload {
         } else {
           vp_token = {
             //TODO: nonce should be initialized correctly
-            nonce: 'NONCE_STRING',
+            nonce: nonce,
```
Comment (nklomp) [inline src/main/AuthenticationRequest.ts:279]: could be just nonce. Als remove the TODO

File: src/main/AuthenticationResponse.ts [hunk 2]
```diff
@@ -268,6 +267,12 @@ async function createSIOPResponsePayload(
     vp_token,
     verifiable_presentations,
   };
+  if (supportedDidMethods.length && resOpts.did) {
```
Comment (nklomp) [inline src/main/AuthenticationResponse.ts:270]: Hmzz, this doesn't sound right.
So if we have a DID we are using a jwk/thumbprint. Should be the other way around

File: src/main/OP.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ export class OP {
 
   public constructor(opts: { builder?: OPBuilder; responseOpts?: AuthenticationResponseOpts; verifyOpts?: VerifyAuthenticationRequestOpts }) {
     this._authResponseOpts = { ...createResponseOptsFromBuilderOrExistingOpts(opts) };
-    this._verifyAuthRequestOpts = { ...createVerifyRequestOptsFromBuilderOrExistingOpts(opts) };
+    this._verifyAuthRequestOpts = { ...createVerifyRequestOptsFromBuilderOrExistingOpts(opts) } as Partial<VerifyAuthenticationRequestOpts>;
```
Comment (nklomp) [inline src/main/OP.ts:35]: What is the reason we need a partial all of a sudden. Sounds like something is off

File: src/main/types/SSI.types.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { JWK } from 'jose';
 
 export interface ResolveOpts {
   resolver?: Resolvable;
+  resolvers?: Map<string, Resolvable>;
```
Comment (nklomp) [inline src/main/types/SSI.types.ts:6]: TBH I am not sure about this approach. Probably we should place a todo here linking to the ticket recently created

**Review by nklomp — CHANGES_REQUESTED** [review]
Some small remarks

File: src/main/AuthenticationRequest.ts (hunk 1 above)
Comment (sksadjad) [reply src/main/AuthenticationRequest.ts:279]: todo removed

File: src/main/AuthenticationResponse.ts (hunk 2 above)
Comment (sksadjad) [reply src/main/AuthenticationResponse.ts:270]: yes, this should be other way around

File: src/main/OP.ts (hunk 3 above)
Comment (sksadjad) [reply src/main/OP.ts:35]: yes, changed some upper method calls expected a partial when it fact they have received a full object. no need for this, removed

File: src/main/types/SSI.types.ts (hunk 4 above)
Comment (sksadjad) [reply src/main/types/SSI.types.ts:7]: added todo for the other ticket

**Review by nklomp — APPROVED** [review]
LGTM

**Pull request merged**